### PR TITLE
Feat/2415405512 dropdown atomic component

### PR DESCRIPTION
**Pull request by pietro-maximoff**
https://sovryn.monday.com/boards/2218344956/pulses/2415405512

### CODE REVIEW

File: src/app/components/Dropdown/index.tsx [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+    } else {
+      maxCenterWidthDropdown = windowWidth;
+    }
+
+    const centerWidthDropdown =
+      widthDropdown > maxCenterWidthDropdown
+        ? maxCenterWidthDropdown
+        : widthDropdown;
+
+    const DropdownPosition = {
```
Comment (tiltom) [inline src/app/components/Dropdown/index.tsx:115]: You can extract it to a function outside the component (with `coords` and `mode` as parameters).

File: src/app/components/Dropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+
+    return DropdownPositionStyles;
+  }, [coords, mode]);
+
+  const classNameComplete = useMemo(
+    () =>
+      classNames(
+        styles.button,
+        isOpen && styles.isOpen,
+        color && styles[color],
```
Comment (tiltom) [inline src/app/components/Dropdown/index.tsx:152]:
```suggestion
        color,
```

File: src/app/components/Dropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+    if (onClose) {
+      onClose();
+    }
+  }, [onClose]);
+
+  useOnClickOutside([buttonRef, dropdownRef], useClickedOutside);
+
+  useEffect(() => {
+    if (!isOpen) return;
+    const coords = getCoords();
```
Comment (tiltom) [inline src/app/components/Dropdown/index.tsx:169]:
```suggestion
    if (!isOpen) {
       return;
    }
    
    const coords = getCoords();
```

File: src/app/components/Dropdown/index.tsx (hunk 2 above)
Comment (tiltom) [reply src/app/components/Dropdown/index.tsx:152]: You can make this adjustment once you adjust `DropdownColor` enum.

File: src/app/components/Dropdown/index.tsx [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        ></span>
+      </button>
+
+      {isOpen && (
+        <Portal target="body">
+          <div
+            className={classNames(
+              styles.dropdown,
+              dropdownClassName,
+              color && styles[color],
```
Comment (tiltom) [inline src/app/components/Dropdown/index.tsx:200]:
```suggestion
              color,
```

File: src/app/hooks/useOnClickOutside.ts [hunk 5]
```diff
@@ -1,31 +1,32 @@
...
+      const { target } = event;
+      if (target && target instanceof HTMLElement) {
+        const shouldIgnoreByRef = ignoreClicksInsideRefs.some(ref =>
+          ref.current?.contains(target),
+        );
 
-      // Do nothing if clicking ref's element or descendent elements
-      if (!el || el.contains(event.target as Node)) {
-        return;
+        !shouldIgnoreByRef && callback(event);
```
Comment (tiltom) [inline src/app/hooks/useOnClickOutside.ts:21]:
```suggestion
        if(!shouldIgnoreByRef) {
           callback(event);
        }
```

**Review by tiltom — CHANGES_REQUESTED** [review]

File: src/app/components/Dropdown/index.module.scss [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    &.isOpen {
+        @apply tw-rounded-b-none;
+    }
+
+    & .iconArrow {
+        @apply tw-w-4 tw-h-4 tw-ml-2 tw-transition-transform tw-bg-contain tw-bg-center tw-bg-no-repeat;
+        background-image: url("data:image/svg+xml,%3Csvg width='23' height='15' xmlns='http://www.w3.org/2000/svg' xmlns:xlink='http://www.w3.org/1999/xlink' xmlns:svgjs='http://svgjs.com/svgjs' viewBox='0 0.399 23 14.203' fill='%23e8e8e8'%3E%3Csvg xmlns='http://www.w3.org/2000/svg' width='23' height='15' viewBox='0.75 0.837 21.5 13.276'%3E%3Cpath paint-order='stroke fill markers' fill-rule='evenodd' d='M19.724.837L11.5 9.043 3.276.837.75 3.363l10.75 10.75 10.75-10.75L19.724.837z'/%3E%3C/svg%3E%3C/svg%3E");
+    }
+}
+.dropdown {
```
[the remark this answers is not in the document]
Comment (ShynRou) [reply src/app/components/Dropdown/index.module.scss]: It is correct this way, as the element this belongs to also applies the className prop. Without using a module.scss class you cannot override the css rules properly neither with tailwind nor normal classes passed into the classNames property.
This is an issue we have on multiple already existing components.

**Review by ShynRou — APPROVED** [review]

**Review by creed-victor — APPROVED** [review]

**Review by tiltom — APPROVED** [review]

**Pull request merged**